### ISSUE TITLE
[ANDROID] RNCalendarEvents.authorizationStatus() returns "denied" even if the user select "Never ask me again" before

### ISSUE THREAD

**Issue by fatihergin** [wmcmahan/react-native-calendar-events#222]
RNCalendarEvents.authorizationStatus() returns "denied" even if the user denied it with selecting the "Never ask me again" checkbox before. It's not returns "restricted" as expected.

### Environment
It's tested with Xiaomi Mi8 (Peanut), Samsung Note 8 (Oreo), Motorola Moto G 4 (Nougat).

```
export async function authorizeForCalendar(): Promise<RNCalendarEventPermissionLevel> {
  let status: RNCalendarEventPermissionLevel = await RNCalendarEvents.authorizationStatus();
  console.info('permission', `previous status${status}`);

  if (status !== 'authorized') {
    status = await RNCalendarEvents.authorizeEventStore();
  }
  console.info('permission', `current status${status}`);

  return status;
}
```

If the user denied this permission with selecting the "Never ask me again" checkbox, "status = await RNCalendarEvents.authorizeEventStore();" line is not returns anything. Only first log is printed in this case.

### Expected Behavior
It should return 'restricted' if the user denied that permission permanently. We can't know the user's former decision to redirect it to the settings to be able to enable permissions manually.

**Comment by Notcracker** [comment]
Having the same problem

**Comment by webtaculars** [comment]
Added a [PR](https://github.com/wmcmahan/react-native-calendar-events/pull/273) to fix this.

**Comment by ldco2016** [comment]
Would any of you know why in my case, the status is always `undetermined`? Somehow that would cause my iOS app to crash until I added `|| status === "undetermined"` like so:

```
export async function createCalendarEvent(event) {
  const status = await RNCalendarEvents.authorizationStatus();
  console.log(status);
  if (status === "authorized" || status === "undetermined") {
    addToCalendar(event);
  } else {
    RNCalendarEvents.authorizeEventStore()
      .then(auth => {
        // handle status
        if (auth === "authorized" || auth === "undetermined") {
          addToCalendar(event);
        }
      })
      .catch(() => {
        alert("This app needs calendar access");
      });
  }
}
```

**Comment by MoOx** [comment]
Closed by 192be03da4c60926055036fae923bf89fa117d5e
